File: models/Appointement.js
import mongoose from 'mongoose';
const { Schema, model } = mongoose;

const AppointementSchema = new Schema({
  
    patient: { type: Schema.Types.ObjectId, ref: 'user' },
    doctor: { type: Schema.Types.ObjectId, ref: 'user' },
    user: { type: Schema.Types.ObjectId, ref: 'user' },
    date: {
        type: Date,
        default: Date.now()
    },
    time: {
        type: String,
    },
    day: {
        type: String,

    },
    upcoming:{
        type: Boolean,
    },
    canceled:{
        type: Boolean,
    },
    statuss: {
        type: String,
    }

},
    {
        timestamp: true
    }
);

export default mongoose.model('appointement', AppointementSchema);

File: lib/models/appointement.dart
import 'package:psychoday/models/user.dart';

class Appointement {
  String? id;
  User? patient;
  User? user;
  User? doctor;
  String? schedule;
  String? date;
  String? time;
  String? day;
  bool? upcoming;
  bool? canceled;
  String? statuss;

  Appointement(this.patient, this.user,this.doctor, this.date, this.time, this.day,
      this.upcoming, this.canceled);
  Appointement.AppointementWithFourParameter(
      {this.doctor, this.date, this.time, this.day, this.statuss});
  Appointement.AppointementWithFourParameterDoctor(
      {this.id,this.user, this.date, this.time, this.day});
}
